src/test/fixtures/fake-client-applications-store.ts

```diff
@@ -78,4 +79,8 @@ export default class FakeClientApplicationsStore
         await this.delete(details.appName);
         return this.bulkUpsert([details]);
     }
+
+    getApplicationOverview(appName: string): Promise<IApplicationOverview> {
```

Comment:
if we had this implemented we could base our testing strategy on service level tests without db which are faster and write only 1 e2e tests that is slow

Reply:
I feel like all of the complexity is in database query. If we are using fake store, we are not testing that complexity.